### ISSUE TITLE
[recursor, dnssec] bla.test-ns.bogus.internet.nl validates as Insecure; expect Bogus

### ISSUE THREAD

**Issue by giganteous** [PowerDNS/pdns#4574]
bla.test-ns.bogus.internet.nl should be bogus.
- a DS record is supplied by the nameservers of internet.nl for test-ns.bogus.internet.nl
- the records in test-ns.bogus.internet.nl are not signed

PowerDNS sees only a proof that bogus.internet.nl does not exist.

Tail of the trace:
```
....
Oct 17 09:45:45 bogus.internet.nl: got negative caching indication for 'bogus.internet.nl|DS'
Oct 17 09:45:45 bogus.internet.nl: status=noerror, other types may exist, but we are done (have negative SOA) (have aa bit) 
Oct 17 09:45:45 validating internet.nl/SOA with 1 sigs
Oct 17 09:45:45 signature by key with tag 28409 was valid
Oct 17 09:45:45 Validated internet.nl/SOA
Oct 17 09:45:45 validating l7pppcnp87n1qsiooboh8aujgkvn9m06.internet.nl/NSEC3 with 1 sigs
Oct 17 09:45:45 signature by key with tag 28409 was valid
Oct 17 09:45:45 Validated l7pppcnp87n1qsiooboh8aujgkvn9m06.internet.nl/NSEC3
Oct 17 09:45:45 got 0 records for DS query of which 0 valid 
Oct 17 09:45:45 No DS for bogus.internet.nl, now look for a secure denial
Oct 17 09:45:45 Do have: internet.nl/SOA
Oct 17 09:45:45 Do have: l7pppcnp87n1qsiooboh8aujgkvn9m06.internet.nl/NSEC3
Oct 17 09:45:45         1 0 5 d9b0a3a053b6809c M6NFRKSTGV5RFG06R8T00Q9MU402FG2B
Oct 17 09:45:45         query hash: l7pppcnp87n1qsiooboh8aujgkvn9m06
Oct 17 09:45:45 Denies existence of DS (not opt-out)
Oct 17 09:45:45 ! state = Insecure, now have 0 keys 
Oct 17 09:45:45 Answer to bla.test-ns.bogus.internet.nl|A for 127.0.0.1:57903 validates as Insecure
Oct 17 09:45:45 6 [1/1] answer to question 'bla.test-ns.bogus.internet.nl|A': 1 answers, 1 additional, took 4 packets, 21.066 ms, 0 throttled, 0 timeouts, 0 tcp connections, rcode=0
```

The full trace is available on https://gist.github.com/giganteous/906af1dc3e34062b62c2a4681aba66b3 .

**Comment by pieterlexis** [comment]
#4389 fixes this issue

**Comment by pieterlexis** [comment]
Fix is merged and backport to 4.0.x is pending.
